pkg/trustedresources/verify_test.go

```diff
@@ -0,0 +1,367 @@
+			Namespace: "default"}}
+	kubeclient := fakek8s.NewSimpleClientset(secret)
+
+	secretref := fmt.Sprintf("%sdefault/verification-secrets", keyReference)
+
+	ctx = test.SetupTrustedResourceConfig(ctx, secretref, config.EnforceResourceVerificationMode)
+
+	unsignedTask := test.GetUnsignedTask("test-task")
+
+	signedTask, err := test.GetSignedTask(unsignedTask, signer, "signed")
+	if err != nil {
+		t.Fatal("fail to sign task", err)
+	}
+
+	tamperedTask := signedTask.DeepCopy()
+	tamperedTask.Annotations["random"] = "attack"
+
+	tcs := []struct {
+		name    string
+		task    v1beta1.TaskObject
+		wantErr bool
+	}{{
+		name:    "Signed Task Passes Verification",
+		task:    signedTask,
+		wantErr: false,
+	}, {
+		name:    "Tampered Task Fails Verification with tampered content",
+		task:    tamperedTask,
+		wantErr: true,
+	}, {
+		name:    "Unsigned Task Fails Verification without signature",
+		task:    unsignedTask,
+		wantErr: true,
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			err := VerifyTask(ctx, tc.task, kubeclient)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("verifyTaskRun() get err %v, wantErr %t", err, tc.wantErr)
+			}
+		})
+	}
+
+}
+
+func TestPrepareObjectMeta(t *testing.T) {
+	unsigned := test.GetUnsignedTask("test-task").ObjectMeta
+
+	signed := unsigned.DeepCopy()
+	signed.Annotations = map[string]string{SignatureAnnotation: "tY805zV53PtwDarK3VD6dQPx5MbIgctNcg/oSle+MG0="}
+
+	signedWithLabels := signed.DeepCopy()
+	signedWithLabels.Labels = map[string]string{"label": "foo"}
+
+	signedWithExtraAnnotations := signed.DeepCopy()
+	signedWithExtraAnnotations.Annotations["kubectl-client-side-apply"] = "client"
+	signedWithExtraAnnotations.Annotations["kubectl.kubernetes.io/last-applied-configuration"] = "config"
+
+	tcs := []struct {
+		name       string
+		objectmeta *metav1.ObjectMeta
+		expected   metav1.ObjectMeta
+		wantErr    bool
+	}{{
+		name:       "Prepare signed objectmeta without labels",
+		objectmeta: signed,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Annotations: map[string]string{},
+		},
+		wantErr: false,
+	}, {
+		name:       "Prepare signed objectmeta with labels",
+		objectmeta: signedWithLabels,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Labels:      map[string]string{"label": "foo"},
+			Annotations: map[string]string{},
+		},
+		wantErr: false,
+	}, {
+		name:       "Prepare signed objectmeta with extra annotations",
+		objectmeta: signedWithExtraAnnotations,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Annotations: map[string]string{},
+		},
+		wantErr: false,
+	}, {
+		name:       "Fail preparation without signature",
+		objectmeta: &unsigned,
+		expected: metav1.ObjectMeta{
+			Name:        "test-task",
+			Namespace:   namespace,
+			Annotations: map[string]string{"foo": "bar"},
+		},
+		wantErr: true,
```

Comment:
apologies, I gave you some feedback about `wantErr` in the context of the e2e tests but that would apply to the rest of these tests as well - I strongly recommend separating out the error cases into their own tests for improved readability and maintainability

Reply:
sure! Thanks for reminding me of those tests. Will do!